libs/net/cluster.py

```
@@ -29,3 +30,23 @@ def ipv6_supported_cluster() -> bool:
+
+
+def cluster_vlan_iterator() -> Generator[int]:
+    """Yield VLAN IDs from the cluster config one at a time.
+
+    The underlying VLAN list is read from py_config once and cached. Each call
+    returns a fresh iterator so every fixture invocation starts from the beginning.
+    Raises ValueError when all VLANs have been consumed.
+    """
+    vlans = _cluster_vlans()
+    yield from vlans
+    raise ValueError(f"vlans list is exhausted. Current list size is {len(vlans)} and all vlans are in use.")
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)